src/parseTree/nodes/node.java

```diff
@@ -0,0 +1,5 @@
+package src.parseTree.nodes;
+
+public interface node {
+    public void addChild(Object child);
+}
```

Comment:
If we're doing this, and I can understand the appeal, there should also be a verify statement in there, so that each node can focus on verifying itself to save complexity in the long run

Reply:
I also suggest a `fix` method to properly assign children to their fields for execution